Fix dependencies for server-side deploy script

feerrenrut opened the pull request:
### Link to issue number:

### Summary of the issue:

The server side deploy code ('nvdaAppveyorHook') relies on artifacts, they must be uploaded before the deploy step is run.

`deploy_script` is run before `on_finish`, so the artifacts were missing.

### Description of how this pull request fixes the issue:
Instead, use `artifacts` section, and also upload from `on_failure`.

For ordering of appveyor `yml` phases, see: https://www.appveyor.com/docs/build-configuration/#build-pipeline

### Testing performed:

### Known issues with pull request:

### Change log entry:

Section: New features, Changes, Bug fixes

feerrenrut committed:
Fix server-side deploy script

The server side deploy code ('nvdaAppveyorHook') relies on artifacts,
they must be uploaded first.

deploy_script is run before on_finish, so the artifacts were missing.
Instead, use artifacts section, and also upload from on_failure.

For ordering of appveyor yml phases,
see: https://www.appveyor.com/docs/build-configuration/#build-pipeline

feerrenrut committed:
Add message linking to PR build artifact

We know that the executable is present, since the name comes from the
'output' directory of the build. Though this depends on the artifact
uploading being successful.

feerrenrut commented:
@michaelDCurran This should be ready to merge. It should fix the issues with the deploy process. There are also now messages for unit / system test failures, and direct link to the PR build executable.

michaelDCurran commented on appveyor.yml:
```diff
@@ -230,3 +253,13 @@ deploy_script:
+      $exe = Get-ChildItem -Name output\*.exe
+      if($?){
+        $exeUrl="$appVeyorUrl/api/buildjobs/$env:APPVEYOR_JOB_ID/artifacts/output/$exe"
+        Add-AppveyorMessage "[Build (for testing PR)]($exeUrl)"
```
As much as I get the logic, I feel this does get a bit too noisy. Especially as it will send an email to anyone associated with the pr, and at first glance looks like the build failed. I'd prefer for now not to include this final part.

feerrenrut replied:
I guess what you mean is that you don't want any comment on successful builds. This is configured in the appveyor settings for the project, along with a template for how these "messages" are formatted. It is in the "notifications" section. I can take this out for now until we discuss it further, but incidentally I've noticed PR's get more testing when a link to the build is easy to access. The process of manually pasting the link in is annoying and easy to forget. 

It would be easy to adjust the template to start with a "Success!" as a heading or whatever. For now I'll disable messages on successful builds.

feerrenrut replied:
For the last two messages from appveyor([success](https://github.com/nvaccess/nvda/pull/10604#issuecomment-564158254) and [failure](https://github.com/nvaccess/nvda/pull/10604#issuecomment-564146161)), the notification template looked like this:

```
{{#jobs}}
{{#messages}}
- {{message}}
{{/messages}}
{{/jobs}}


{{#failed}}
See [test results]({{buildUrl}}/tests) for [{{status}} build]({{buildUrl}}) of [commit {{commitId}}]({{commitUrl}})
{{/failed}}


{{#passed}}
See [{{status}} build]({{buildUrl}}) of [commit {{commitId}}]({{commitUrl}})
{{/passed}}
```
This is a "Mustache template" the [appveyor docs in the "customizing  message template" section](https://www.appveyor.com/docs/notifications/#customizing-message-template) have some examples, and a link to a [more formal description of the format](http://mustache.github.io/mustache.5.html). On the same page, but further up there is a description of the [available variables in the "webhook payload" section](https://www.appveyor.com/docs/notifications/#webhook-payload-default).

michaelDCurran reviewed (changes requested):
This all looks good to me, and based on appveyor documentation should work as we expect. However, we won't truely know until we try to make beta2.

feerrenrut committed:
Add more build failure messages.

Adds a message for:
- Unable to install NVDA
- Unit test failure
- System test failure

feerrenrut commented:
I have changed the target of this PR to Beta, and rebased the changes. Assuming a successful CI build, I believe we can merge it.
